agent/setup.go

```diff
@@ -27,13 +30,15 @@ import (
 type BaseDeps struct {
 	Logger          hclog.InterceptLogger
 	TLSConfigurator *tlsutil.Configurator // TODO: use an interface
-	MetricsHandler  MetricsHandler
-	RuntimeConfig   *config.RuntimeConfig
-	Tokens          *token.Store
-	Cache           *cache.Cache
-	AutoConfig      *autoconf.AutoConfig // TODO: use an interface
-	ConnPool        *pool.ConnPool       // TODO: use an interface
-	Router          *router.Router
+	// todo(kit): migrate this to an internal metrics client
+	MetricsClient  *metrics.Metrics
```

Comment:
Will this eventually be some interface type instead of a concrete armon/go-metrics Client?

Reply:
Yep! It's a placeholder same as the other entry in the proxy.